### ISSUE TITLE
jobs occationally don't have internet connecttivity

### ISSUE THREAD

**Issue by tfoote** [ros-infrastructure/buildfarm_deployment#47]
Example failed jobs:
http://54.183.26.131:8080/job/Jrel_release-status-page/1685/console
http://54.183.26.131:8080/job/Jrel_arm_release-status-page/1685/console
http://54.183.26.131:8080/job/Irel_release-status-page/1685/console
http://54.183.26.131:8080/job/Irel_arm_release-status-page/1688/console

```
6. Run Dockerfile - status page

Hide Details
# BEGIN SECTION: Run Dockerfile - status page 19:17:47 + rm -fr /home/jenkins-slave/workspace/Jrel_release-status-page/debian_repo_cache 19:17:47 + rm -fr /home/jenkins-slave/workspace/Jrel_release-status-page/status_page 19:17:47 + mkdir -p /home/jenkins-slave/workspace/Jrel_release-status-page/debian_repo_cache 19:17:47 + mkdir -p /home/jenkins-slave/workspace/Jrel_release-status-page/status_page 19:17:47 + docker run --cidfile=/home/jenkins-slave/workspace/Jrel_release-status-page/docker_generate_status_page/docker.cid --net=host -v /home/jenkins-slave/workspace/Jrel_release-status-page/ros_buildfarm:/tmp/ros_buildfarm:ro -v /home/jenkins-slave/workspace/Jrel_release-status-page/debian_repo_cache:/tmp/debian_repo_cache -v /home/jenkins-slave/workspace/Jrel_release-status-page/status_page:/tmp/status_page status_page_generation 19:17:48 The build file contains the following targets: 19:17:48 - trusty source 19:17:48 - trusty amd64 19:17:48 - trusty i386 19:17:48 - utopic source 19:17:48 - utopic amd64 19:17:48 - utopic i386 19:17:48 - vivid source 19:17:48 - vivid amd64 19:17:48 - vivid i386 19:18:20 Traceback (most recent call last): 19:18:20 File "/usr/lib/python3.4/urllib/request.py", line 1232, in do_open 19:18:20 h.request(req.get_method(), req.selector, req.data, headers) 19:18:20 File "/usr/lib/python3.4/http/client.py", line 1065, in request 19:18:20 self._send_request(method, url, body, headers) 19:18:20 File "/usr/lib/python3.4/http/client.py", line 1103, in _send_request 19:18:20 self.endheaders(body) 19:18:20 File "/usr/lib/python3.4/http/client.py", line 1061, in endheaders 19:18:20 self._send_output(message_body) 19:18:20 File "/usr/lib/python3.4/http/client.py", line 906, in _send_output 19:18:20 self.send(msg) 19:18:20 File "/usr/lib/python3.4/http/client.py", line 841, in send 19:18:20 self.connect() 19:18:20 File "/usr/lib/python3.4/http/client.py", line 819, in connect 19:18:20 self.timeout, self.source_address) 19:18:20 File "/usr/lib/python3.4/socket.py", line 509, in create_connection 19:18:20 raise err 19:18:20 File "/usr/lib/python3.4/socket.py", line 500, in create_connection 19:18:20 sock.connect(sa) 19:18:20 socket.timeout: timed out 19:18:20 19:18:20 During handling of the above exception, another exception occurred: 19:18:20 19:18:20 Traceback (most recent call last): 19:18:20 File "/usr/lib/python3/dist-packages/rosdistro/loader.py", line 48, in load_url 19:18:20 fh = urlopen(url, timeout=timeout) 19:18:20 File "/usr/lib/python3.4/urllib/request.py", line 153, in urlopen 19:18:20 return opener.open(url, data, timeout) 19:18:20 File "/usr/lib/python3.4/urllib/request.py", line 455, in open 19:18:20 response = self._open(req, data) 19:18:20 File "/usr/lib/python3.4/urllib/request.py", line 473, in _open 19:18:20 '_open', req) 19:18:20 File "/usr/lib/python3.4/urllib/request.py", line 433, in _call_chain 19:18:20 result = func(*args) 19:18:20 File "/usr/lib/python3.4/urllib/request.py", line 1258, in http_open 19:18:20 return self.do_open(http.client.HTTPConnection, req) 19:18:20 File "/usr/lib/python3.4/urllib/request.py", line 1235, in do_open 19:18:20 raise URLError(err) 19:18:20 urllib.error.URLError: <urlopen error timed out> 19:18:20 19:18:20 During handling of the above exception, another exception occurred: 19:18:20 19:18:20 Traceback (most recent call last): 19:18:20 File "/usr/lib/python3.4/urllib/request.py", line 1232, in do_open 19:18:20 h.request(req.get_method(), req.selector, req.data, headers) 19:18:20 File "/usr/lib/python3.4/http/client.py", line 1065, in request 19:18:20 self._send_request(method, url, body, headers) 19:18:20 File "/usr/lib/python3.4/http/client.py", line 1103, in _send_request 19:18:20 self.endheaders(body) 19:18:20 File "/usr/lib/python3.4/http/client.py", line 1061, in endheaders 19:18:20 self._send_output(message_body) 19:18:20 File "/usr/lib/python3.4/http/client.py", line 906, in _send_output 19:18:20 self.send(msg) 19:18:20 File "/usr/lib/python3.4/http/client.py", line 841, in send 19:18:20 self.connect() 19:18:20 File "/usr/lib/python3.4/http/client.py", line 819, in connect 19:18:20 self.timeout, self.source_address) 19:18:20 File "/usr/lib/python3.4/socket.py", line 509, in create_connection 19:18:20 raise err 19:18:20 File "/usr/lib/python3.4/socket.py", line 500, in create_connection 19:18:20 sock.connect(sa) 19:18:20 socket.timeout: timed out 19:18:20 19:18:20 During handling of the above exception, another exception occurred: 19:18:20 19:18:20 Traceback (most recent call last): 19:18:20 File "/usr/lib/python3/dist-packages/rosdistro/loader.py", line 48, in load_url 19:18:20 fh = urlopen(url, timeout=timeout) 19:18:20 File "/usr/lib/python3.4/urllib/request.py", line 153, in urlopen 19:18:20 return opener.open(url, data, timeout) 19:18:20 File "/usr/lib/python3.4/urllib/request.py", line 455, in open 19:18:20 response = self._open(req, data) 19:18:20 File "/usr/lib/python3.4/urllib/request.py", line 473, in _open 19:18:20 '_open', req) 19:18:20 File "/usr/lib/python3.4/urllib/request.py", line 433, in _call_chain 19:18:20 result = func(*args) 19:18:20 File "/usr/lib/python3.4/urllib/request.py", line 1258, in http_open 19:18:20 return self.do_open(http.client.HTTPConnection, req) 19:18:20 File "/usr/lib/python3.4/urllib/request.py", line 1235, in do_open 19:18:20 raise URLError(err) 19:18:20 urllib.error.URLError: <urlopen error timed out> 19:18:20 19:18:20 During handling of the above exception, another exception occurred: 19:18:20 19:18:20 Traceback (most recent call last): 19:18:20 File "/usr/lib/python3.4/urllib/request.py", line 1232, in do_open 19:18:20 h.request(req.get_method(), req.selector, req.data, headers) 19:18:20 File "/usr/lib/python3.4/http/client.py", line 1065, in request 19:18:20 self._send_request(method, url, body, headers) 19:18:20 File "/usr/lib/python3.4/http/client.py", line 1103, in _send_request 19:18:20 self.endheaders(body) 19:18:20 File "/usr/lib/python3.4/http/client.py", line 1061, in endheaders 19:18:20 self._send_output(message_body) 19:18:20 File "/usr/lib/python3.4/http/client.py", line 906, in _send_output 19:18:20 self.send(msg) 19:18:20 File "/usr/lib/python3.4/http/client.py", line 841, in send 19:18:20 self.connect() 19:18:20 File "/usr/lib/python3.4/http/client.py", line 819, in connect 19:18:20 self.timeout, self.source_address) 19:18:20 File "/usr/lib/python3.4/socket.py", line 509, in create_connection 19:18:20 raise err 19:18:20 File "/usr/lib/python3.4/socket.py", line 500, in create_connection 19:18:20 sock.connect(sa) 19:18:20 socket.timeout: timed out 19:18:20 19:18:20 During handling of the above exception, another exception occurred: 19:18:20 19:18:20 Traceback (most recent call last): 19:18:20 File "/usr/lib/python3/dist-packages/rosdistro/loader.py", line 48, in load_url 19:18:20 fh = urlopen(url, timeout=timeout) 19:18:20 File "/usr/lib/python3.4/urllib/request.py", line 153, in urlopen 19:18:20 return opener.open(url, data, timeout) 19:18:20 File "/usr/lib/python3.4/urllib/request.py", line 455, in open 19:18:20 response = self._open(req, data) 19:18:20 File "/usr/lib/python3.4/urllib/request.py", line 473, in _open 19:18:20 '_open', req) 19:18:20 File "/usr/lib/python3.4/urllib/request.py", line 433, in _call_chain 19:18:20 result = func(*args) 19:18:20 File "/usr/lib/python3.4/urllib/request.py", line 1258, in http_open 19:18:20 return self.do_open(http.client.HTTPConnection, req) 19:18:20 File "/usr/lib/python3.4/urllib/request.py", line 1235, in do_open 19:18:20 raise URLError(err) 19:18:20 urllib.error.URLError: <urlopen error timed out> 19:18:20 19:18:20 During handling of the above exception, another exception occurred: 19:18:20 19:18:20 Traceback (most recent call last): 19:18:20 File "/tmp/ros_buildfarm/scripts/status/build_release_status_page.py", line 34, in <module> 19:18:20 main() 19:18:20 File "/tmp/ros_buildfarm/scripts/status/build_release_status_page.py", line 30, in main 19:18:20 args.cache_dir, args.output_dir, copy_resources=args.copy_resources) 19:18:20 File "/tmp/ros_buildfarm/ros_buildfarm/status_page.py", line 50, in build_release_status_page 19:18:20 dist = get_cached_distribution(index, rosdistro_name) 19:18:20 File "/usr/lib/python3/dist-packages/rosdistro/__init__.py", line 146, in get_cached_distribution 19:18:20 cache = get_distribution_cache(index, dist_name) 19:18:20 File "/usr/lib/python3/dist-packages/rosdistro/__init__.py", line 172, in get_distribution_cache 19:18:20 yaml_gz_str = load_url(url, skip_decode=True) 19:18:20 File "/usr/lib/python3/dist-packages/rosdistro/loader.py", line 58, in load_url 19:18:20 return load_url(url, retry=retry - 1, retry_period=retry_period, timeout=timeout) 19:18:20 File "/usr/lib/python3/dist-packages/rosdistro/loader.py", line 58, in load_url 19:18:20 return load_url(url, retry=retry - 1, retry_period=retry_period, timeout=timeout) 19:18:20 File "/usr/lib/python3/dist-packages/rosdistro/loader.py", line 59, in load_url 19:18:20 raise URLError(str(e) + ' (%s)' % url) 19:18:20 urllib.error.URLError: <urlopen error <urlopen error timed out> (http://ros.org/rosdistro/jade-cache.yaml.gz)> 19:18:20 Build step 'Execute shell' marked build as failure
```

A slightly different error: 
```
6. Run Dockerfile - status page

Hide Details
# BEGIN SECTION: Run Dockerfile - status page 19:15:12 + rm -fr /home/jenkins-slave/workspace/Irel_arm_release-status-page/debian_repo_cache 19:15:12 + rm -fr /home/jenkins-slave/workspace/Irel_arm_release-status-page/status_page 19:15:12 + mkdir -p /home/jenkins-slave/workspace/Irel_arm_release-status-page/debian_repo_cache 19:15:12 + mkdir -p /home/jenkins-slave/workspace/Irel_arm_release-status-page/status_page 19:15:12 + docker run --cidfile=/home/jenkins-slave/workspace/Irel_arm_release-status-page/docker_generate_status_page/docker.cid --net=host -v /home/jenkins-slave/workspace/Irel_arm_release-status-page/ros_buildfarm:/tmp/ros_buildfarm:ro -v /home/jenkins-slave/workspace/Irel_arm_release-status-page/debian_repo_cache:/tmp/debian_repo_cache -v /home/jenkins-slave/workspace/Irel_arm_release-status-page/status_page:/tmp/status_page status_page_generation 19:15:14 The build file contains the following targets: 19:15:14 - trusty source 19:15:14 - trusty armhf 19:15:42 Traceback (most recent call last): 19:15:42 File "/usr/lib/python3.4/urllib/request.py", line 1232, in do_open 19:15:42 h.request(req.get_method(), req.selector, req.data, headers) 19:15:42 File "/usr/lib/python3.4/http/client.py", line 1065, in request 19:15:42 self._send_request(method, url, body, headers) 19:15:42 File "/usr/lib/python3.4/http/client.py", line 1103, in _send_request 19:15:42 self.endheaders(body) 19:15:42 File "/usr/lib/python3.4/http/client.py", line 1061, in endheaders 19:15:42 self._send_output(message_body) 19:15:42 File "/usr/lib/python3.4/http/client.py", line 906, in _send_output 19:15:42 self.send(msg) 19:15:42 File "/usr/lib/python3.4/http/client.py", line 841, in send 19:15:42 self.connect() 19:15:42 File "/usr/lib/python3.4/http/client.py", line 819, in connect 19:15:42 self.timeout, self.source_address) 19:15:42 File "/usr/lib/python3.4/socket.py", line 509, in create_connection 19:15:42 raise err 19:15:42 File "/usr/lib/python3.4/socket.py", line 500, in create_connection 19:15:42 sock.connect(sa) 19:15:42 socket.timeout: timed out 19:15:42 19:15:42 During handling of the above exception, another exception occurred: 19:15:42 19:15:42 Traceback (most recent call last): 19:15:42 File "/usr/lib/python3/dist-packages/rosdistro/loader.py", line 48, in load_url 19:15:42 fh = urlopen(url, timeout=timeout) 19:15:42 File "/usr/lib/python3.4/urllib/request.py", line 153, in urlopen 19:15:42 return opener.open(url, data, timeout) 19:15:42 File "/usr/lib/python3.4/urllib/request.py", line 461, in open 19:15:42 response = meth(req, response) 19:15:42 File "/usr/lib/python3.4/urllib/request.py", line 571, in http_response 19:15:42 'http', request, response, code, msg, hdrs) 19:15:42 File "/usr/lib/python3.4/urllib/request.py", line 493, in error 19:15:42 result = self._call_chain(*args) 19:15:42 File "/usr/lib/python3.4/urllib/request.py", line 433, in _call_chain 19:15:42 result = func(*args) 19:15:42 File "/usr/lib/python3.4/urllib/request.py", line 676, in http_error_302 19:15:42 return self.parent.open(new, timeout=req.timeout) 19:15:42 File "/usr/lib/python3.4/urllib/request.py", line 455, in open 19:15:42 response = self._open(req, data) 19:15:42 File "/usr/lib/python3.4/urllib/request.py", line 473, in _open 19:15:42 '_open', req) 19:15:42 File "/usr/lib/python3.4/urllib/request.py", line 433, in _call_chain 19:15:42 result = func(*args) 19:15:42 File "/usr/lib/python3.4/urllib/request.py", line 1258, in http_open 19:15:42 return self.do_open(http.client.HTTPConnection, req) 19:15:42 File "/usr/lib/python3.4/urllib/request.py", line 1235, in do_open 19:15:42 raise URLError(err) 19:15:42 urllib.error.URLError: <urlopen error timed out> 19:15:42 19:15:42 During handling of the above exception, another exception occurred: 19:15:42 19:15:42 Traceback (most recent call last): 19:15:42 File "/usr/lib/python3.4/urllib/request.py", line 1232, in do_open 19:15:42 h.request(req.get_method(), req.selector, req.data, headers) 19:15:42 File "/usr/lib/python3.4/http/client.py", line 1065, in request 19:15:42 self._send_request(method, url, body, headers) 19:15:42 File "/usr/lib/python3.4/http/client.py", line 1103, in _send_request 19:15:42 self.endheaders(body) 19:15:42 File "/usr/lib/python3.4/http/client.py", line 1061, in endheaders 19:15:42 self._send_output(message_body) 19:15:42 File "/usr/lib/python3.4/http/client.py", line 906, in _send_output 19:15:42 self.send(msg) 19:15:42 File "/usr/lib/python3.4/http/client.py", line 841, in send 19:15:42 self.connect() 19:15:42 File "/usr/lib/python3.4/http/client.py", line 819, in connect 19:15:42 self.timeout, self.source_address) 19:15:42 File "/usr/lib/python3.4/socket.py", line 509, in create_connection 19:15:42 raise err 19:15:42 File "/usr/lib/python3.4/socket.py", line 500, in create_connection 19:15:42 sock.connect(sa) 19:15:42 socket.timeout: timed out 19:15:42 19:15:42 During handling of the above exception, another exception occurred: 19:15:42 19:15:42 Traceback (most recent call last): 19:15:42 File "/usr/lib/python3/dist-packages/rosdistro/loader.py", line 48, in load_url 19:15:42 fh = urlopen(url, timeout=timeout) 19:15:42 File "/usr/lib/python3.4/urllib/request.py", line 153, in urlopen 19:15:42 return opener.open(url, data, timeout) 19:15:42 File "/usr/lib/python3.4/urllib/request.py", line 455, in open 19:15:42 response = self._open(req, data) 19:15:42 File "/usr/lib/python3.4/urllib/request.py", line 473, in _open 19:15:42 '_open', req) 19:15:42 File "/usr/lib/python3.4/urllib/request.py", line 433, in _call_chain 19:15:42 result = func(*args) 19:15:42 File "/usr/lib/python3.4/urllib/request.py", line 1258, in http_open 19:15:42 return self.do_open(http.client.HTTPConnection, req) 19:15:42 File "/usr/lib/python3.4/urllib/request.py", line 1235, in do_open 19:15:42 raise URLError(err) 19:15:42 urllib.error.URLError: <urlopen error timed out> 19:15:42 19:15:42 During handling of the above exception, another exception occurred: 19:15:42 19:15:42 Traceback (most recent call last): 19:15:42 File "/tmp/ros_buildfarm/scripts/status/build_release_status_page.py", line 34, in <module> 19:15:42 main() 19:15:42 File "/tmp/ros_buildfarm/scripts/status/build_release_status_page.py", line 30, in main 19:15:42 args.cache_dir, args.output_dir, copy_resources=args.copy_resources) 19:15:42 File "/tmp/ros_buildfarm/ros_buildfarm/status_page.py", line 50, in build_release_status_page 19:15:42 dist = get_cached_distribution(index, rosdistro_name) 19:15:42 File "/usr/lib/python3/dist-packages/rosdistro/__init__.py", line 146, in get_cached_distribution 19:15:42 cache = get_distribution_cache(index, dist_name) 19:15:42 File "/usr/lib/python3/dist-packages/rosdistro/__init__.py", line 172, in get_distribution_cache 19:15:42 yaml_gz_str = load_url(url, skip_decode=True) 19:15:42 File "/usr/lib/python3/dist-packages/rosdistro/loader.py", line 58, in load_url 19:15:42 return load_url(url, retry=retry - 1, retry_period=retry_period, timeout=timeout) 19:15:42 File "/usr/lib/python3/dist-packages/rosdistro/loader.py", line 58, in load_url 19:15:42 return load_url(url, retry=retry - 1, retry_period=retry_period, timeout=timeout) 19:15:42 File "/usr/lib/python3/dist-packages/rosdistro/loader.py", line 65, in load_url 19:15:42 return contents.decode('utf-8') 19:15:42 UnicodeDecodeError: 'utf-8' codec can't decode byte 0x8b in position 1: invalid start byte
```

**Comment by dirk-thomas** [comment]
Some more examples from the last fifteen minutes:
* http://54.183.26.131:8080/job/Ibin_uT64__libsiftfast__ubuntu_trusty_amd64__binary/8/
* http://54.183.26.131:8080/job/Ibin_uT64__jsk_tilt_laser__ubuntu_trusty_amd64__binary/8/
* http://54.183.26.131:8080/job/Ibin_uT64__jsk_topic_tools__ubuntu_trusty_amd64__binary/8/
* http://54.183.26.131:8080/job/Ibin_uT64__jsk_footstep_msgs__ubuntu_trusty_amd64__binary/8/
* http://54.183.26.131:8080/job/Jsrc_uV__euslisp__ubuntu_vivid__source/2/

**Comment by dirk-thomas** [comment]
Btw. the content of the first post is barely readable since most of the output is in a single line.

**Comment by tfoote** [comment]
We had it a little different tonight that the squid docker instance had no PID, but docker still thought it was running. Restarting docker fixed that issue.

**Comment by tfoote** [comment]
I think this was fixed in #57
